prov/shm/src/smr_progress.c

```diff
@@ -138,8 +139,11 @@ static int smr_progress_resp_entry(struct smr_ep *ep, struct smr_resp *resp,
 			"unidentified operation type\n");
 	}
 
-	if (fastlock_tryacquire(&peer_smr->lock))
-		return -FI_EAGAIN;
+	if (strcmp(smr_name(peer_smr), smr_name(ep->region))) {
```

Comment:
This is an expensive comparison to insert into the fast path.  I would find a different way to compare.  Maybe compare the shared memory region pointers?

Reply:
Would the pointers be the same if we mapped our own region again?

Reply:
Answer is apparently yes! Updated.

Reply:
Ugh nevermind, I switched the if to only lock if they are the same 🙄 The comparison fails. Any other ideas?